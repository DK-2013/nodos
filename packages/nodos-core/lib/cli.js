import _ from 'lodash';
import yargs from 'yargs';
// import '@babel/register';
import { nodos } from '.';
import log from './logger';

export default async (args = process.argv.slice(2), options = {}) => {
  const {
    exitProcess = true,
    projectRoot = process.cwd(),
    container = {},
    done = _.noop, // FIXME: hack for testing purposes
  } = options;
  const parser = yargs(args);
  parser.exitProcess(exitProcess);
  // parser.fail(done.fail);
  parser.demandCommand();
  parser.recommendCommands();
  parser.strict();
  parser.showHelpOnFail(true);
  const nodosItem = _.get(container, 'nodos', nodos);
  const app = await nodosItem(projectRoot);
  await app.boot();
  // .epilog(help.trim())
  // .example(example.trim())
  // console.log(app);
  // console.log(app.commandBuilders);
  const commands = app.commands.map(build => build({ app, container, done }));
  console.log(commands);
  commands.forEach(c => parser.command(c));
  await parser.argv;
  return app;
};
