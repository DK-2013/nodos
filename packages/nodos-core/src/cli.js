import _ from 'lodash';
import yargs from 'yargs';
import { nodos } from '.';
import log from './logger';

export default async (args = process.argv.slice(2), options = {}) => {
  const {
    exitProcess = true,
    projectRoot = process.cwd(),
    container = {},
    done = _.noop, // FIXME: hack for testing purposes
  } = options;
  const parser = yargs(args);
  parser.exitProcess(exitProcess);
  // parser.fail(done.fail);
  parser.demandCommand();
  parser.recommendCommands();
  parser.strict();
  parser.showHelpOnFail(true);
  const nodosItem = _.get(container, 'nodos', nodos);
  const app = await nodosItem(projectRoot);
  // .epilog(help.trim())
  // .example(example.trim())
  // console.log(app);
  // console.log(app.commandBuilders);
  const commands = app.commandBuilders.map(build => build({ app, container, done }));
  console.log(commands);
  commands.forEach(c => parser.command(c));
  return parser.argv;
};
