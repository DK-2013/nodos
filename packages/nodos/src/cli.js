import _ from 'lodash';
import yargs from 'yargs';
import commandsBuilder from './commands';

export default (args = process.argv.slice(2), options = {}) => {
  const { exitProcess = true, container = {}, done = _.noop } = options;
  const commands = commandsBuilder(container, done);
  const parser = yargs(args);
  parser.exitProcess(exitProcess);
  // parser.fail(done.fail);
  parser.demandCommand();
  parser.recommendCommands();
  parser.strict();
  parser.showHelpOnFail(true);
  parser.option('projectRoot', {
    alias: 'r',
    default: process.cwd(),
  });
  // .epilog(help.trim())
  // .example(example.trim())
  commands.forEach(command => parser.command(command));
  return parser.argv;
};
