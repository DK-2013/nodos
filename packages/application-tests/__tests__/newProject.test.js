import { promises as fsp } from 'fs';
import path from 'path';
import { nodos } from '@nodosjs/core';
import { runNew } from '@nodosjs/cli';
import execa from 'execa';

// TODO: development пришлось отрубить так как запускается вебпкак который не умеет каскадно смотреть либы
// SassError: Can't find stylesheet to import.
const envs = ['production', 'test'];

let projectRoot;

beforeAll(async () => {
  const dir = path.join(__dirname, '..', '__applications__');
  await fsp.mkdir(dir, { recursive: true });
  const apps = await fsp.readdir(dir);
  apps.sort();
  const oldAppPaths = apps.reverse().slice(1).map((app) => path.join(dir, app));
  await Promise.all(oldAppPaths.map((appPath) => fsp.rmdir(appPath, { recursive: true })));

  const appName = `nodos-${Date.now().toString()}`;
  projectRoot = path.join(dir, appName);

  const options = { exitProcess: false, args: ['new', appName] };
  await runNew(dir, options);
});

test.each(envs)('start server', async (env) => {
  const app = await nodos(projectRoot, env);
  await app.initApp();
  await app.listen();
  await app.close();
  expect(true).toBe(true);
});

test.each(envs)('check cli', async (env) => {
  const options = {
    cwd: projectRoot,
    env: { NODE_ENV: env },
  };
  const result = await execa('npx', ['nodos', 'routes'], options);
  expect(result).not.toBeNull();
  expect(result.stdout).toMatchSnapshot();
});
