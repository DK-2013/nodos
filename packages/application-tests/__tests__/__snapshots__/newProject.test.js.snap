// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check cli 1`] = `""`;

exports[`check cli 2`] = `""`;

exports[`check cli 3`] = `""`;
